src/Umbraco.Infrastructure/Runtime/CoreInitialServices.cs

```diff
@@ -62,15 +62,17 @@
 using Umbraco.Web.Trees;
 using TextStringValueConverter = Umbraco.Core.PropertyEditors.ValueConverters.TextStringValueConverter;
 
-namespace Umbraco.Core.Runtime
+namespace Umbraco.Infrastructure.Runtime
 {
-    // core's initial composer composes before all core composers
-    [ComposeBefore(typeof(ICoreComposer))]
-    public class CoreInitialComposer : ComponentComposer<CoreInitialComponent>
+    public static class CoreInitialServices
     {
-        public override void Compose(IUmbracoBuilder builder)
+        public static IUmbracoBuilder AddCoreInitialServices(this IUmbracoBuilder builder)
         {
-            base.Compose(builder);
+            builder.AddNotificationHandler<UmbracoApplicationStarting, EssentialDirectoryCreator>();
+
+            builder.Services.AddSingleton<ManifestWatcher>();
+            builder.AddNotificationHandler<UmbracoApplicationStarting, ManifestWatcher>(factory => factory.GetRequiredService<ManifestWatcher>());
```

Comment:
I reaaaaaaally don't like this as it opens up even dispatching to abuse. 

All event handler should be transient and resolved directly by the service provider.